daml-script/daml3/Daml/Script/Questions/Submit.daml

```diff
@@ -1,6 +1,7 @@
 -- Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
 -- SPDX-License-Identifier: Apache-2.0
-{-# LANGUAGE CPP #-}
+{-# LANGUAGE UndecidableInstances #-}
```

Comment:
Sadly needed due to the `IsSubmitOptions` and `IsParties` schenanigans

Reply:
going back to this, would it be too bad to replace `instance IsParties p => IsSubmitOptions p` with decidable instances that mirror the ones for `IsParties`? it looks like we just have five of those, namely:

* `instance IsParties Party`
* `instance IsParties (Optional Party)`
* `instance IsParties [Party]`
* `instance IsParties (NonEmpty Party)`
* `instance IsParties (Set Party)`

Reply:
Yeah that probably cleaner, i might also remove the Optional instance